rnd/autogpt_builder/src/components/history.ts

```diff
@@ -0,0 +1,44 @@
+// history.ts
+
+type Action = {
+    type: 'ADD_NODE' | 'DELETE_NODE' | 'ADD_EDGE' | 'DELETE_EDGE' | 'UPDATE_NODE' | 'MOVE_NODE' | 'UPDATE_INPUT' | 'UPDATE_NODE_POSITION';
+    payload: any;
+    undo: () => void;
+    redo: () => void;
+};
+
+class History {
```

Comment:
I'm not sure if it would help but would providing a clear() and a getHistoryState() method here provide value to you?

If so you can add something like:

```typescript
  clear() {
    this.past = [];
    this.future = [];
  }

  getHistoryState() {
    return {
      past: [...this.past],
      future: [...this.future],
    };
  }
  ```